hazelcast/src/test/java/com/hazelcast/cp/CPGroupAvailabilityListenerTest.java

```diff
@@ -135,4 +140,75 @@ public void majorityLost(CPGroupAvailabilityEvent event) {
             majorityLatch.countDown();
         }
     }
+
+    static class GracefulShutdownAvailabilityListener
+            implements CPGroupAvailabilityListener {
+        private final AtomicInteger availabilityDecreased;
+        private final AtomicInteger majorityLost;
+
+        private final Set<UUID> membersShutdown;
+
+        GracefulShutdownAvailabilityListener() {
+            availabilityDecreased = new AtomicInteger();
+            majorityLost = new AtomicInteger();
+            membersShutdown = Collections.synchronizedSet(new HashSet<>());
```

Comment:
`ConcurrentHashMap.newKeySet()` - ?

Reply:
Sure, why not -- saves a few characters I suppose.